client/src/app/components/add-rule/add-rule.component.html

```diff
@@ -149,39 +149,46 @@ <h3>Datapoint</h3>
             *ngIf="currentRuleConditionDataType"
           >
             <mat-label>Value</mat-label>
-            <ng-container [ngSwitch]="currentRuleConditionDataType">
+            <ng-container [ngSwitch]="true">
+              <mat-select
+                *ngSwitchCase="currentRuleConditionValues && currentRuleConditionValues.length > 0"
+                [(ngModel)]="currentRuleConditionCompareValue"
+                matInput
+                name="rule-value"
+                required
+              >
+                <mat-option
+                  *ngFor="let value of currentRuleConditionValues"
+                  value="{{ value }}"
+                  >{{ value }}</mat-option
+                >
+              </mat-select>
               <input
-                *ngSwitchCase="'number'"
+                *ngSwitchCase="currentRuleConditionDataType === 'number'"
```

Comment:
Would you have to add the !currentRuleConditionValue check in here as well? In case you have a "number" enum you would otherwise run into the same double input issue as with string, no?

Reply:
Good catch! Added.